library/core/src/slice/mod.rs

```diff
@@ -78,6 +78,11 @@ pub use index::check_range;
 #[lang = "slice"]
 #[cfg(not(test))]
 impl<T> [T] {
+    const MAX_LEN_BYTES_BOUND: usize =
+        [isize::MAX as usize, usize::MAX][(mem::size_of::<T>() == 0) as usize];
+    const MAX_LEN_BOUND: usize =
+        Self::MAX_LEN_BYTES_BOUND / [mem::size_of::<T>(), 1][(mem::size_of::<T>() == 0) as usize];
```

Comment:
I feel like we just stabilized if in consts, perhaps we could use that here instead of indexing into arrays?